week2/RO_flowsheet_with_ERD_build.ipynb

```diff
@@ -0,0 +1,469 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the connectivity using Arcs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "84719040",
+   "metadata": {},
+   "source": [
+    "### Step 2f: Expand Arcs\n",
+    "\n",
+    "<center><img src=\"graphics/workflow-step2f.png\" width=\"600\" /></center>\n",
+    "\n",
+    "After establishing these connection, the arcs are expanded to create the constraints between all the state variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "96faa069",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use the TransformationFactory to expand the arcs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4250afc",
+   "metadata": {},
+   "source": [
+    "### Step 2g: Add Variables, Constraints, and Objectives\n",
+    "\n",
+    "<center><img src=\"graphics/workflow-step2g.png\" width=\"600\" /></center>\n",
+    "\n",
+    "The last step before model specification is to add any additional variables, constraints, or objectives that are desired on the model. This is an optional step, but allows the user to create custom relationships on the fly.\n",
+    "\n",
+    "For demonstration purposes, below we create a variable and constraint that defines the RO flux in LMH. Uncomment the variable and constraint below, then create an objective function that will maximize the RO flux.\n",
+    "\n",
+    "<details>\n",
+    "  <summary>Click the arrow for a hint!</summary>\n",
+    "    \n",
+    "The syntax for creating an objective function is: `m.fs.UserDefinedName = Objective(expr=m.fs.VariableName, sense=\"maximize\" or \"minimize\")`\n",
+    "</details>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "60036788",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# m.fs.RO.flux_LMH = Var(\n",
+    "#     initialize=20.0,\n",
+    "#     bounds=(0, 45),\n",
+    "#     units=pyunits.liter / (pyunits.m**2 * pyunits.hr),\n",
+    "#     doc=\"Water flux through the membrane in LMH\",\n",
+    "# )\n",
+    "\n",
+    "# m.fs.RO.flux_LMH_constraint = Constraint(\n",
+    "#     expr=m.fs.RO.flux_LMH\n",
+    "#     == pyunits.convert(\n",
+    "#         m.fs.RO.mixed_permeate[0].flow_vol_phase[\"Liq\"] / m.fs.RO.area,\n",
+    "#         to_units=pyunits.liter / (pyunits.m**2 * pyunits.hr),\n",
+    "#     )\n",
+    "# )\n",
+    "\n",
+    "# Write an objective function that maximizes the RO flux"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cf81e3a6",
+   "metadata": {},
+   "source": [
+    "## Step 3: Specify Model\n",
+    "\n",
+    "<center><img src=\"graphics/workflow-step3.png\" width=\"600\" /></center>\n",
+    "\n",
+    "As before, we must provide parameter values for all the degrees of freedom on the model. Note that the `product` block is fully specified by its connected arcs, so we don't need to specify any additional degrees of freedom.\n",
```

Comment:
We need to provide the specifications here, otherwise people won't know where to start, and people may surely put in values that will lead to infeasibility.